Framework/Core/include/Framework/DataInputDirector.h

```diff
@@ -103,10 +106,12 @@ struct DataInputDirector {
 
   // getters
   DataInputDescriptor* getDataInputDescriptor(header::DataHeader dh);
+  int getNumberInputDescriptors() { return mdataInputDescriptors.size(); }
+
   std::unique_ptr<TTreeReader> getTreeReader(header::DataHeader dh, int counter, int numTF, std::string treeName);
-  std::tuple<TFile*, std::string> getFileFolder(header::DataHeader dh, int counter, int numTF);
   TTree* getDataTree(header::DataHeader dh, int counter, int numTF);
-  int getNumberInputDescriptors() { return mdataInputDescriptors.size(); }
+  int64_t getTimeFrameNumber(header::DataHeader dh, int counter, int numTF);
+  std::tuple<TFile*, std::string> getFileFolder(header::DataHeader dh, int counter, int numTF);
```

Comment:
Please do not use std::tuple, but create a small struct with meaningful names for the members (no need to have getters).

Reply:
done